aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -248,6 +248,15 @@
    */
   public static final String S3_WRITE_TAGS_PREFIX = "s3.write.tags.";
 
+  /**
+   * Used by {@link S3FileIO} to tag objects when deleting. To set, we can pass a catalog property.
+   * <p>
+   * For more details, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html
```

Comment:
it's a bit sudden to link this feature to lifecycle management. We should describe that, "when this config is set, objects are tagged with the configured key-value pairs instead of being hard-deleted. This is considered a soft-delete, because users are able to configure tag-based object lifecycle policy at bucket level to transition objects to different tiers"

Reply:
Thanks for the feedback. I have made the changes.

Reply:
Thanks, for the feedback. I have made the changes.